File: io-hotmoka-network-thin-client/io-hotmoka-network-thin-client-kt/src/main/kotlin/io/hotmoka/network/thin/client/models/updates/ClassTagModel.kt
package io.hotmoka.network.thin.client.models.updates

import io.hotmoka.network.thin.client.models.values.TransactionReferenceModel

/**
 * The model of the class tag of an object.
 */
class ClassTagModel(
        /**
         * The name of the class of the object.
         */
        val className: String,
        /**
         * The transaction that installed the jar from where the class has been loaded.
         */
        val jar: TransactionReferenceModel
)

File: io-hotmoka-network-thin-client/io-hotmoka-network-thin-client-js/src/models/updates/ClassTagModel.ts
import {TransactionReferenceModel} from "../values/TransactionReferenceModel";

/**
 * The model of the class tag of an object.
 */
export class ClassTagModel {
    /**
     * The name of the class of the object.
     */
    className: string

    /**
     * The transaction that installed the jar from where the class has been loaded.
     */
    jar: TransactionReferenceModel

    constructor(className: string, jar: TransactionReferenceModel) {
        this.className = className
        this.jar = jar
    }

}